pinot-common/src/main/java/org/apache/pinot/common/utils/URIUtils.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.utils;
+
+import java.io.File;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.util.StringJoiner;
+
+
+public class URIUtils {
+  private URIUtils() {
+  }
+
+  /**
+   * Returns the URI for the given base path and optional parts, appends the local (file) scheme to the URI if no
+   * scheme exists. All the parts will be appended to the base path with the file separator.
+   */
+  public static URI getUri(String basePath, String... parts) {
```

Comment:
The second param is not a list of segments. It's just the path. Maybe we should use a better name?

Reply:
Oh you changed it before I submitted the comments. :)